<launch>
  <rosparam command="load" file="$(find mapping_kit)/config/r3live_config.yaml" />

  <node pkg="image_transport" type="republish" name="uncompress_l2d" args="compressed in:=/intel/color/image_raw_10hz out:=/intel/color/image_raw_10hz_uncompressed" />

  <node name="lidar2depth" pkg="lidar2depth" type="lidar2depth" output="screen">
    <remap from="color" to="/intel/color/image_raw_10hz_uncompressed"/>
    <remap from="lidar" to="cloud_registered"/>
    <remap from="color_sync" to="livox/color_sync/image_rect"/>
    <remap from="depth" to="livox/depth"/>
    <param name="max_cloud_size" value="500000"/>
    <param name="map_frame" value="world"/>
    <param name="lidar_frame" value="aft_mapped"/>
    <param name="filter_kernel" value="1"/>
    <param name="depth_width" value="640"/>
    <param name="depth_height" value="480"/>
    <param name="frame_rate" value="5"/>
  </node>

  <node name="test_projection" pkg="lidar2depth" type="test_projection.py" output="screen">
    <remap from="image" to="livox/color_sync/image_rect"/>
    <remap from="depth" to="livox/depth"/>
  </node>  
</launch>
